core/src/trezor/crypto/slip39.py

```diff
@@ -105,6 +106,287 @@ def bits_to_words(n: int) -> int:
 """The index of the share containing the digest of the shared secret."""
 
 
+"""
+# Keyboard functions
+"""
+
+KEYBOARD_FULL_MASK = const(0x1FF)
+"""All buttons are allowed. 9-bit bitmap all set to 1."""
+
+
+def compute_mask(prefix: str) -> int:
+    if not prefix:
+        return KEYBOARD_FULL_MASK
+    return slip39.compute_mask(int(prefix))
+
+
+def button_sequence_to_word(prefix: str) -> str:
+    if not prefix:
+        return ""
+    return slip39.button_sequence_to_word(int(prefix))
+
+
+"""
+# External API
+"""
+
+MAX_SHARE_COUNT = const(16)
+"""The maximum number of shares that can be created."""
+
+DEFAULT_ITERATION_EXPONENT = const(1)
+
+
+class ParsedMnemonic:
```

Comment:
Consider renaming this class to "Share" and moving the encode_mnemonic() and decode_mnemonic() functions into this class and calling them to_mnemonic() and from_mnemonic().

Reply:
I really disliked "ParsedMnemonic", "Share" is way better. Done.

-------

I've considered putting the functions into the class, but I don't like it. We would have to do something like this for example:

```python
share = Share()
share.from_mnemonic(...)
```

and then we would need to check the values etc. Or we would need two constructors ideally :). I gave it a go here, if you are interested: https://github.com/trezor/trezor-firmware/commit/1533276ba5d801e767c9243d66a8d643935e91cc